pkg/bloomutils/ring.go

```diff
@@ -14,20 +14,47 @@ import (
 	v1 "github.com/grafana/loki/pkg/storage/bloom/v1"
 )
 
-type InstanceWithTokenRange struct {
-	Instance           ring.InstanceDesc
-	MinToken, MaxToken uint32
+type integer interface {
+	~uint | ~uint32 | ~uint64 | ~int | ~int32 | ~int64
 }
 
-func (i InstanceWithTokenRange) Cmp(token uint32) v1.BoundsCheck {
-	if token < i.MinToken {
+type Range[T integer] struct {
```

Comment:
Maybe `GetInstanceWithTokenRange` can be generic and return Range[T]. So we can map tokens to all ok the Integer key-spaces.

Reply:
Same for `NewInstanceSortMergeIterator`